spark/src/main/scala/ai/chronon/spark/JoinUtils.scala

```diff
@@ -47,11 +47,16 @@ object JoinUtils {
     } else {
       Seq()
     }
-    val scanQuery = range.genScanQuery(joinConf.left.query,
-                                       joinConf.left.table,
-                                       fillIfAbsent = Map(tableUtils.partitionColumn -> null) ++ timeProjection) +
-      limit.map(num => s" LIMIT $num").getOrElse("")
-    val df = tableUtils.sql(scanQuery)
+    val query = joinConf.left.query
+    val partitionColumn = tableUtils.getPartitionColumn(query)
+    val (scanQuery, df) = range.scanQueryStringAndDf(
+      query,
+      joinConf.left.table,
+      appendRawQueryString = limit.map(num => s" LIMIT $num").getOrElse(""),
+      fillIfAbsent = Map(partitionColumn -> null) ++ timeProjection,
+      partitionColOpt = Some(partitionColumn),
+      renamePartitionCol = true
+    )
```

Comment:
nit: add negative spacing for readability (basically newlines around 4-5 line blocks that logically do one thing)
```suggestion

    val (scanQuery, df) = range.scanQueryStringAndDf(
      query,
      joinConf.left.table,
      appendRawQueryString = limit.map(num => s" LIMIT $num").getOrElse(""),
      fillIfAbsent = Map(partitionColumn -> null) ++ timeProjection,
      partitionColOpt = Some(partitionColumn),
      renamePartitionCol = true
    )
    
```